src/contract/fungible.rs

```diff
@@ -478,13 +465,12 @@ pub struct ConcealedValue {
     /// Pedersen commitment to the original [`FungibleState`].
     pub commitment: PedersenCommitment,
     /// Range proof for the [`FungibleState`] not exceeding type boundaries.
-    #[commit_encode(skip)]
     pub range_proof: RangeProof,
```

Comment:
Every time the contract_id method is called, a new ID is generated.

After investigating, I noticed that the ConcealedValue::commit method always generates a new default value for RangeProof. So if we ignore the field at the time of serialization, the error no longer occurs.

But I'm not sure this is the solution.


```suggestion
    #[strict_type(skip)]
    pub range_proof: RangeProof,
```

Reply:
No, you miss the whole point here.

We MUST NOT commit to the rangeproofs - but we MUST encode and keep them.

Right now, since we do not have bulletproofs in place, we do not have any rangeproofs, and I deliberatly generate noise instead - and it was right since this allows us to detect the issue, that we actually commit to the rangeproofs somehow!

I do not see any code doing that and can't understand how it is possible. Since the commitments is the most security-critical part of RGB, I believe that they have to be absolutelly transparent in the way how the commitments are actually created. So I will have to continue my effort on making the system generating automatic description of how the commitments happens (with Vesper). Also, I think we need to do a ~100% test coverage of all commitments workflows before going live with v0.11.